### PR TITLE
fix(wrapped): narrow bash splitter to && / ||; exclude shell keywords

### PR DESCRIPTION
**Pull request by csansoon**
## summary

After PR #3117 shipped, real Wrapped reports surfaced strange "top commands" entries — `BUILD)"`, `warning:"`, `api[_-]key\`, `service.name\`, `set(evals_sched)`, `do`, `"`. Root cause: the segment splitter regex broke commands on `|` and `;` even when those characters appeared *inside* shell-quoted strings or regex patterns.

A `grep "service.name\|api[_-]key" file` was split at the `\|` mid-regex into junk segments. A `psql -c "alter …; set(evals_sched);"` was split at the `;` mid-SQL. A `for x in *; do …; done` was split at the `;`, leaving `do` as the prefix of a downstream segment.

The principled fix would be a quote-aware shell tokeniser. The pragmatic fix is to be much more selective about which operators we split on.

## what changed

**Bash splitter narrowed to `&&` and `||` only.** `|` and `;` are now treated as content. They live inside their segment.

**Shell control-flow keywords added to the plumbing set.** With the splitter change, a `for x in *; do …; done` command becomes one segment with prefix `for` — that prefix needs to be excluded too. Adds `for`, `while`, `until`, `if`, `case`, `select`, `function` plus the after-`;` keywords (`do`, `done`, `then`, `else`, `elif`, `fi`, `esac`, `in`) for completeness. The TS classifier set (`BASH_PLUMBING_PREFIXES`) and the SQL literal (`BASH_PLUMBING_PREFIXES_SQL`) move together.

## trade-offs

Piped-after commands (`pnpm test | tail`) and bare-`;`-chained commands (`cd ..; pnpm test`) no longer get their own segment — the first command owns the whole thing. The right-hand-side of a pipe is almost always a plumbing tool we'd exclude anyway (`tail`, `head`, `wc`, `xargs`), and Claude rarely uses bare `;` for chaining (it uses `&&`). So `commandsRun` drops only marginally and bucket routing is unchanged for the common case.

Edge case not handled: a literal `&&` or `||` inside a quoted string would still produce a false split. A proper quote-aware tokeniser would catch that but isn't worth it for a display surface.

## test plan

- [x] 558 tests pass on `@domain/spans`. 2 new cases: block-start keywords (`for`/`while`/`if`/`case`) excluded; after-`;` keywords (`do`/`done`/`then`/`else`/`elif`/`fi`/`esac`/`in`) excluded.
- [x] `pnpm --filter @platform/db-clickhouse --filter @domain/spans typecheck` clean.
- [x] `pnpm biome check packages/domain/spans/src packages/platform/db-clickhouse/src` clean.
- [ ] Trigger a Wrapped post-merge for the projects that surfaced the issue. Confirm `BUILD)"`, `set(evals_sched)`, `do`, `api[_-]key\` etc. no longer appear in any workspace's top commands.

Refs #3117

### CODE REVIEW

File: packages/platform/db-clickhouse/src/repositories/claude-code-span-reader.ts [hunk 1]
```diff
@@ -46,16 +46,27 @@ const FILE_PATH_TOOLS = ["Read", "Edit", "Write", "NotebookEdit", "MultiEdit"] a
...
+// Cost: piped-after commands and bare-`;`-chained commands don't get
+// their own segment. In practice the RHS of a pipe is almost always a
+// plumbing tool (`| tail`, `| grep`, `| wc`) we already exclude, and
+// Claude rarely uses bare `;` for chaining (it uses `&&`). So the net
+// change to `commandsRun` and bucket counts is small, and the
+// junk-token elimination is worth far more.
+//
+// Subshells (`$(…)`, backticks) are still not handled — proper handling
+// would need a real shell tokeniser, not worth it.
+const BASH_SEGMENT_REGEX = "\\\\s*(?:&&|\\\\|\\\\|)\\\\s*"
```
[the remark this answers is not in the document]
Comment (csansoon) [reply packages/platform/db-clickhouse/src/repositories/claude-code-span-reader.ts:70]: Good catch — fixed in `6b3c677d9`. Extracted a pure TS `splitBashSegments` mirror of the CH `BASH_SEGMENT_REGEX` and added 12 new tests:

- **6 direct splitter cases** asserting the regex behaviour: `&&` and `||` split; `grep "foo\|bar"`, `psql -c "alter …; set(x); …"`, `for x in *; do …; done`, and shell pipes (`pnpm test | tail`) do NOT split.
- **6 full-pipeline cases** that walk raw bash strings through splitter → tokeniser → classifier and assert the resulting bucket sequence. The canonical mixed `cd .. && grep "foo\|bar" file && pnpm test | tail` is pinned at exactly 3 segments routing to `[excluded, search, bash]`. If the SQL regex ever drifts back to also splitting on `|` or `;`, the TS spec would catch it.

Same pattern we use for `extractBashTokens` — the TS function is the unit-testable contract, the CH SQL mirrors it.

**Review by andresgutgon — APPROVED** [review]
